internal/v1/handler_blueprints.go

```diff
@@ -191,22 +199,44 @@ func (h *Handlers) ComposeBlueprint(ctx echo.Context, id openapi_types.UUID) err
 	}
 	composeResponses := make([]ComposeResponse, 0, len(blueprint.ImageRequests))
 	clientId := ClientId("api")
-	for _, imageRequest := range blueprint.ImageRequests {
-		composeRequest := ComposeRequest{
-			Customizations:   &blueprint.Customizations,
-			Distribution:     blueprint.Distribution,
-			ImageRequests:    []ImageRequest{imageRequest},
-			ImageName:        &blueprintEntry.Name,
-			ImageDescription: &blueprintEntry.Description,
-			ClientId:         &clientId,
+	if requestBody.ImageTypes == nil {
+		for _, imageRequest := range blueprint.ImageRequests {
+			composeRequest := ComposeRequest{
+				Customizations:   &blueprint.Customizations,
+				Distribution:     blueprint.Distribution,
+				ImageRequests:    []ImageRequest{imageRequest},
+				ImageName:        &blueprintEntry.Name,
+				ImageDescription: &blueprintEntry.Description,
+				ClientId:         &clientId,
+			}
+
+			composesResponse, err := h.handleCommonCompose(ctx, composeRequest, &blueprintEntry.VersionId)
+			if err != nil {
+				return err
+			}
+			composeResponses = append(composeResponses, composesResponse)
 		}
-		composesResponse, err := h.handleCommonCompose(ctx, composeRequest, &blueprintEntry.VersionId)
-		if err != nil {
-			return err
+	} else {
+		for _, imageRequest := range blueprint.ImageRequests {
+			if !slices.Contains(*requestBody.ImageTypes, imageRequest.ImageType) {
```

Comment:
Can't we just adjust this condition instead of having two branches (if/else) for basically the same thing?
Seeing the same code twice repeat after few lines scares me.
```suggestion
			if requestBody.ImageTypes != nil && !slices.Contains(*requestBody.ImageTypes, imageRequest.ImageType) {
```